.github/build-docs.sh

```diff
@@ -6,13 +6,14 @@ set -o pipefail
 
 # python dependencies
 pip3 install -U "Pygments>=2.4.1"
+pip3 install -U "sphinx>=7.2.3"
```

Comment:
Why the version constraint in `requirements-dev.txt` doesn't work?

Reply:
Keep those version constraints in `requirements-dev.txt`.

Reply:
> Why the version constraint in `requirements-dev.txt` doesn't work?

Get the following error in my local machine while building the docs.

```bash
Exception occurred:
  File "/usr/local/lib/python3.10/dist-packages/breathe/project.py", line 116, in __init__
    self._default_build_dir = os.path.dirname(app.doctreedir.rstrip(os.sep))
AttributeError: 'PosixPath' object has no attribute 'rstrip'
```

Related issue:
https://github.com/sphinx-doc/sphinx/issues/11605